Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -59,6 +59,35 @@ def check_python_framework_links(lib)
     EOS
   end
 
+  def check_python_virtualenv
+    return if formula_requires_python?(formula) && default_python_is_system_python?
+    framework = formula.libexec/".Python"
+    return unless framework.symlink?
+    return unless framework.realpath.to_s.start_with?("/System")
+    <<-EOS.undent
+      virtualenv created against system Python
+      This formula created a virtualenv using system Python.
+      Please add `depends_on :python` to the formula.
+    EOS
+  end
+
+  def check_python_shebangs
+    return unless formula.bin.directory?
+    return if formula_requires_python?(formula) && default_python_is_system_python?
+    system_python_shebangs = formula.bin.children.select do |bin|
+      (bin.open { |f| f.read(32) }).start_with?("#!/usr/bin/python")
```

Comment:
Why 32? If it's the string length can you make it a variable and get the length dynamically that way?